cypress/e2e/laboratory-preflight-script.cy.ts

```diff
@@ -186,6 +197,66 @@ throw new TypeError('Test')`,
 });
 
 describe('Execution', () => {
+  it('result.request.headers are added to the graphiql request base headers', () => {
+    const headers = {
+      foo: { name: 'foo', value: 'bar' },
+    };
+    cy.dataCy(selectors.buttonToggle).click();
+    cy.dataCy(selectors.buttonModal).click();
+    setEditorScript(`lab.request.headers.append('${headers.foo.name}', '${headers.foo.value}')`);
+    cy.dataCy(selectors.modal.buttonSubmit).click();
+    cy.intercept({
+      method: 'POST',
+      headers: {
+        [headers.foo.name]: headers.foo.value,
+      },
+    }).as('post');
+    cy.get(selectors.graphiql.buttonExecute).click();
+    cy.wait('@post');
+  });
+
+  it('result.request.headers take precedence over graphiql request base headers', () => {
+    // --- Pre Assert Integrity Check: make sure the header we think we're overriding is actually there.
+    const baseHeaders = {
+      accept: {
+        name: 'accept',
+        value: 'application/json, multipart/mixed',
+      },
+    };
+    cy.intercept({
+      method: 'POST',
+      headers: {
+        [baseHeaders.accept.name]: baseHeaders.accept.value,
+      },
+    }).as('integrityCheck');
+    cy.get(selectors.graphiql.buttonExecute).click();
+    cy.wait('@integrityCheck');
+    // ---
+
+    const preflightHeaders = {
+      accept: {
+        name: 'accept',
+        value: 'application/graphql-response+json; charset=utf-8, application/json; charset=utf-8',
+      },
+    };
+    cy.dataCy(selectors.buttonToggle).click();
+    cy.dataCy(selectors.buttonModal).click();
+    setEditorScript(
+      `lab.request.headers.append('${preflightHeaders.accept.name}', '${preflightHeaders.accept.value}')`,
+    );
+    cy.dataCy(selectors.modal.buttonSubmit).click();
+    cy.intercept({
+      method: 'POST',
+      headers: {
+        [preflightHeaders.accept.name]: preflightHeaders.accept.value,
+      },
+    }).as('post');
+    cy.get(selectors.graphiql.buttonExecute).click();
+    cy.wait('@post');
```

Comment:
I found a bug wherein:
- if GraphiQL is executed
- and then the preflight script is enabled, setup with a headers altering script that overrides some base headers
- the effect of the preflight script on request is nil (the base headers are not overridden)

If the preflight headers are something not in base headers, then it works.

I'm not sure what's going on, investigating.